### PR TITLE
Merge PgQuery and QueryBuilder...

### PR DESCRIPTION
**Pull request by diogob**
... or let's remove some code!

The idea here is just to make the internal API simpler.
I tried to reduce the interfaces, and eliminate functions that where not used, or move the top level ones to where clauses.
I still want to simplify the QueryBuilder interface and document it, but I'll send in another PR for more convenient review and to avoid conflicts.

### CODE REVIEW

**Comment by ruslantalpa** [comment]
:+1:

File: src/PostgREST/QueryBuilder.hs [hunk 1]
```diff
@@ -1,52 +1,76 @@
...
-  find (\r -> s == (tableSchema . relTable) r && t1 == (tableName . relTable) r && t2 == (tableName . relFTable) r) allRelations
+type PStmt = H.Stmt P.Postgres
+instance Monoid PStmt where
+  mappend (B.Stmt query params prep) (B.Stmt query' params' prep') =
+    B.Stmt (query <> query') (params <> params') (prep && prep')
+  mempty = B.Stmt "" empty True
+type StatementT = PStmt -> PStmt
 
-addRelations :: Schema -> [Relation] -> Maybe ApiRequest -> ApiRequest -> Either Text ApiRequest
+addRelations :: Text -> [Relation] -> Maybe ApiRequest -> ApiRequest -> Either Text ApiRequest
```
Comment (begriffs) [inline src/PostgREST/QueryBuilder.hs]: No more `Schema`? I thought it was a nice type alias.

File: src/PostgREST/QueryBuilder.hs [hunk 2]
```diff
@@ -84,11 +196,11 @@ requestToQuery schema (Node (Select colSelects tbls conditions ord, (mainTbl, _)
     tblSchema tbl = if tbl == sourceSubqueryName then "" else schema
     qi = QualifiedIdentifier (tblSchema mainTbl) mainTbl
     toQi t = QualifiedIdentifier (tblSchema t) t
-    query = Data.Text.unwords [
-      ("WITH " <> intercalate ", " withs) `emptyOnNull` withs,
-      "SELECT ", intercalate ", " (map (pgFmtSelectItem qi) colSelects ++ selects),
-      "FROM ", intercalate ", " (map (fromQi . toQi) tbls),
-      ("WHERE " <> intercalate " AND " ( map (pgFmtCondition qi ) conditions )) `emptyOnNull` conditions,
+    query = T.unwords [
```
Comment (begriffs) [inline src/PostgREST/QueryBuilder.hs]: Nice abbrev.

File: src/PostgREST/QueryBuilder.hs (hunk 1 above)
Comment (diogob) [inline src/PostgREST/QueryBuilder.hs]: sorry, problems rebasing :\

**Pull request merged**